dll/win32/shell32/folders/CRecycleBin.cpp

```diff
@@ -171,16 +123,94 @@ BOOL WINAPI CBSearchRecycleBin(IN PVOID Context, IN HANDLE hDeletedFile)
     return ret;
 }
 
-static PIDLRecycleStruct * _ILGetRecycleStruct(LPCITEMIDLIST pidl)
+/*
+ * Recycle Bin folder
+ */
+
+HRESULT CRecyclerExtractIcon_CreateInstance(
+    LPCITEMIDLIST pidl, REFIID riid, LPVOID * ppvOut)
 {
-    LPPIDLDATA pdata = _ILGetDataPointer(pidl);
+    PIDLRecycleStruct *pFileDetails = _ILGetRecycleStruct(pidl);
 
-    if (pdata && pdata->type == 0x00)
-        return (PIDLRecycleStruct*) & (pdata->u.crecycle);
+    // Try to obtain the file
+    SEARCH_CONTEXT Context;
+    Context.pFileDetails = pFileDetails;
+    Context.bFound = FALSE;
 
-    return NULL;
+    if (pFileDetails == NULL) goto error;
```

Comment:
So, could be done before initializing the structure (see comment above).

Reply:
I think yes, idk if ros is built with C99 or newer, if yes then it can be possible, i will test this tomorrow

Reply:
Since this is a C++ file there is no problem.